app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java

```diff
@@ -300,6 +321,32 @@ public void onComplete() {}
         };
     }
 
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.menu_item_removeWatched:
+                RemoveWatchedStreams remover = new RemoveWatchedStreams();
+
+                remover.onPreExecute();
+                removeWatchedWorker = Flowable.just(playlistManager.getPlaylistStreams(playlistId).blockingFirst())
```

Comment:
Before this line you should add this code to dispose `removeWatchedWorker` if it is not null:
```java
if (removeWatchedWorker != null) {
    removeWatchedWorker.dispose();
}
```

Reply:
Fixed in commit [81bcac5](https://github.com/TeamNewPipe/NewPipe/pull/3065/commits/81bcac5d2800b14a4e1917c9ffbfe4064b5fb967)